extensions/resource-deployment/src/interfaces.ts

```diff
@@ -45,27 +55,45 @@ export interface DialogInfo {
 
 export interface DialogTabInfo {
 	title: string;
-	sections: DialogSectionInfo[];
+	sections: SectionInfo[];
+	labelWidth?: string;
+	inputWidth?: string;
 }
 
-export interface DialogSectionInfo {
+export interface SectionInfo {
 	title: string;
-	fields: DialogFieldInfo[];
+	fields?: FieldInfo[]; // Use this if the dialog is not wide. All fields will be displayed in one column, label will be placed on top of the input component.
+	rows?: RowInfo[]; // Use this for wide dialog or wizard. label will be placed to the left of the input component.
+	labelWidth?: string;
+	inputWidth?: string;
+	labelOnLeft?: boolean; // label position. true: label on the left side of field, false: label on top of field, default is false.
+	collapsible?: boolean;
```

Comment:
Any reason you don't want to use an enum instead of labelOnLeft boolean?

---
In reply to: [323883357](https://github.com/microsoft/azuredatastudio/pull/7183#discussion_r323883357) [](ancestors = 323883357)